statsmodels/sandbox/stats/multicomp.py

```diff
@@ -778,6 +786,117 @@ def plot_simultaneous(self, comparison_name=None, ax=None, figsize=(10,6),
         ax1.set_ylabel(ylabel if ylabel is not None else '')
         return fig
 
+class GamesHowellResults(object):
+    """Results from Games Howell test, with additional
+
+    Can also compute and plot additional post-hoc evaluations using this
+    results class.
+
+    Parameters
+    ----------
+    mc_object : MultiComparison instance
+    gh_results : np.array
+        nine dimensional array with columns group 1, group 2, mean differences,
+        standard errors, q statistic, degrees of freedom, adjusted p value,
+        lower limit of confidence interval, upper limit of confidence interval.
+
+    Attributes
+    ----------
+    data : ndarray
+        independent data samples
+    groups: ndarray
+        group labels corresponding to each data point
+    groupsunqiue: ndarray
+        group labels with no repetition
+    groupmeans: ndarray
+        mean of data by group
+
+    Notes
+    -----
+    These attributes contain information about the data from the
+    MultiComparison instance: data, groups, groupsunique, groupmeans.
+    """
+    def __init__(self, mc_object, gh_results):
+
+        self._multicomp = mc_object
+        self._results = gh_results
+        # Taken out of _multicomp for ease of access for unknowledgeable users
+        self.data = self._multicomp.data
+        self.groups = self._multicomp.groups
+        self.groupsunique = self._multicomp.groupsunique
+        self.groupmeans = self._multicomp.groupstats.groupmean
+
+    def summary(self, alpha=0.05):
+        """Summary table that can be printed
+        """
+        results_table = SimpleTable(self._results, headers=self._results.dtype.names)
+        results_table.title = f"Multiple Comparison of Means - Games Howell, FWER={alpha}"
+        return results_table
+
+    def summary_df(self):
+        """Summary table as a dataframe
+        """
+        return pd.DataFrame(self._results)
+
+    def connecting_letters(self, sig_level = 0.05):
```

Comment:
Is this important here? It is the only use of nx. Doesn't seem to be related to the test.

Reply:
This is a good discussion to have.  You are correct, we can add the statistical test without this function. However, both [JMP]((and we should discuss the name of this also if we keep it and where it should live, like be abstracted so users can use it on both Tukey and Games Howell results). ) and [R](https://cran.r-project.org/web/packages/multcompView/multcompView.pdf) have such a report for Tukey and it makes using the results much much easier.  

And if we agree to keep it, we should discuss the name of this function, since I'm not sure we can use the same name as what JMP calls it and discuss where it should live --- in this implementation it can only be applied to Games Howell results, but with a bit of refactoring of the Tukey code and some abstraction here it could be used to visualize results from either test and is really helpful for such visualization.

I use networkx because the algorithm for this is to compute cliques in a graph.  To remove that dependency and keep this visualization of the results would require re-implementing that algorithm.

Reply:
It possibly could be added as a soft dependency.  Would need to consult @josef-pkt . In this case, networkx would only be imported in the function where it is needed in a try, except ImportError, and if not available a "friendly" error message would be given to users to tell them that networkx is required but not available.

Reply:
I haven't looked at the details yet. I've never seen a graph representation for this.
I have a function (in a PR ?) that assigns letters to individual groups, for a standard linear display (list of letters as grouping).

Maybe we can just return some structure that users can feed directly to `nx` instead of us doing it.
Otherwise we make it optional and build graph if nx is available.

Reply:
@josef-pkt can you point me to that function.  I think it is doing something fundamentally different from this, but without looking I'm not sure.   The issue here is the algorithm --- I use a dataframe to display the results/return it to the user.  I'm using `networkx` to find the cliques in the graph that tell you how the groups are connected.  I need to look again at what the raw return is from the `find_cliques` function in `nx` is to see how user friendly, or not that might be.  

I personally, like @bashtage's suggestion to use a try, except, ImportError with a graceful/friendly fail to help users and understand if that's not considered an option.  Note that `plot_simultaneous` in `TukeyHSDResults` requires `matplotlib` and has none of this --- I'm not saying we should follow that pattern, rather to say we should set the pattern here we want in other cases, like that, for the future.